test/streaming.test.ts

```diff
@@ -32,10 +32,14 @@ if (isNodeJS()) {
         .subscribe(resolve);
     });
     await delay(5000);
-    await conn
-      .sobject('Account')
-      .create({ Name: `My New Account #${Date.now()}` });
+    const accountName = `My New Account #${Date.now()}`;
+
+    await conn.sobject('Account').create({ Name: accountName });
+
     const msg = await msgArrived;
+    assert.ok(isObject(msg.sobject));
+    assert.ok(msg.sobject.Name === accountName);
```

Comment:
sometimes it would fail like this:
https://github.com/jsforce/jsforce/actions/runs/6897338638/job/18765305207#step:8:552

added some assertions to ensure the msg had the correct account record. Then jest retries will help with intermittent failures from API.